JustSaying/AwsTools/QueueCreation/RegionResourceCache.cs

```diff
@@ -1,33 +1,36 @@
 using System.Collections.Generic;
 
 namespace JustSaying.AwsTools.QueueCreation
-{
-    public class RegionResourceCache<T> : Dictionary<string, Dictionary<string, T>>, IRegionResourceCache<T>
+{
+    public sealed class RegionResourceCache<T>
     {
+        private readonly Dictionary<string, Dictionary<string, T>> _regionsData

```

Comment:
Maybe we should use ConcurrentDictionary instead?

Reply:
I had some weird issues when trying to parallelalalalalalize subscription creation which came down to this class and specifically reading / writing to the dictionary

Reply:
![image](https://38.media.tumblr.com/41c4beab5f5a24af6a8eabd40d395faf/tumblr_nt1fa21Mea1udb050o1_400.gif)

Reply:
So we currently have `Dictionary<string, Dictionary<string, T>>` 
Do we need `ConcurrentDictionary<string, Dictionary<string, T>>` or `ConcurrentDictionary<string, ConcurrentDictionary<string, T>>` ?

Reply:
Something like IDictionary<string, IDictionary<string, T>>, then the actual implementations can be concurrent dictionaries. The type of the field doesn’t have to actually mention ConcurrentDictionary and can just leverage the co/contra/whichever-variance.

Reply:
Something like IDictionary<string, IDictionary<string, T>>, then the actual implementations can be concurrent dictionaries. The type of the field doesn’t have to actually mention ConcurrentDictionary and can just leverage the co/contra/whichever-variance.